src/Datadog.Trace.ClrProfiler.Managed/Integrations/SqlCommandIntegration.cs

```diff
@@ -0,0 +1,470 @@
+        /// <returns>The value returned by the instrumented method.</returns>
+        [InterceptMethod(
+            TargetAssemblies = new[] { AdoNetConstants.AssemblyNames.SystemData, AdoNetConstants.AssemblyNames.SystemDataSqlClient },
+            TargetType = SqlCommandTypeName,
+            TargetMethod = AdoNetConstants.MethodNames.ExecuteReader,
+            TargetSignatureTypes = new[] { SqlDataReaderTypeName, AdoNetConstants.TypeNames.CommandBehavior },
+            TargetMinimumVersion = Major4,
+            TargetMaximumVersion = Major4)]
+        public static object ExecuteReader(
+            object command,
+            int behavior,
+            int opCode,
+            int mdToken,
+            long moduleVersionPtr)
+        {
+            Func<object, CommandBehavior, object> instrumentedMethod;
+            var commandBehavior = (CommandBehavior)behavior;
+
+            try
+            {
+                var targetType = command.GetInstrumentedType(SqlCommandTypeName);
+
+                instrumentedMethod =
+                    MethodBuilder<Func<object, CommandBehavior, object>>
+                       .Start(moduleVersionPtr, mdToken, opCode, AdoNetConstants.MethodNames.ExecuteReader)
+                       .WithConcreteType(targetType)
+                       .WithParameters(commandBehavior)
+                       .WithNamespaceAndNameFilters(SqlDataReaderTypeName, AdoNetConstants.TypeNames.CommandBehavior)
+                       .Build();
+            }
+            catch (Exception ex)
+            {
+                Log.ErrorException($"Error resolving {SqlCommandTypeName}.{AdoNetConstants.MethodNames.ExecuteReader}(...)", ex);
+                throw;
+            }
+
+            using (var scope = ScopeFactory.CreateDbCommandScope(Tracer.Instance, command as DbCommand, IntegrationName))
+            {
+                try
+                {
+                    return instrumentedMethod(command, commandBehavior);
+                }
+                catch (Exception ex)
+                {
+                    scope?.Span.SetException(ex);
+                    throw;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Instrumentation wrapper for SqlCommand.ExecuteReader().
+        /// </summary>
+        /// <param name="command">The object referenced by this in the instrumented method.</param>
+        /// <param name="behavior">The <see cref="CommandBehavior"/> value used in the original method call.</param>
+        /// <param name="method">The "method" string used in the original method call.</param>
+        /// <param name="opCode">The OpCode used in the original method call.</param>
+        /// <param name="mdToken">The mdToken of the original method call.</param>
+        /// <param name="moduleVersionPtr">A pointer to the module version GUID.</param>
+        /// <returns>The value returned by the instrumented method.</returns>
+        [InterceptMethod(
```

Comment:
Do we need both of these methods? For .NET Framework it seems redundant but I haven't looked at the implementation of `System.Data.SqlClient` for .NET Core

Reply:
In the `System.Data` assembly included with .NET Framework, `ExecuteReader()` and `ExecuteReader(CommandBehavior)` both call into `ExecuteReader(CommandBehavior, string)`, so that seems like a better "single point of interception."

In the `System.Data.SqlClient` assembly you get via the NuGet package, `ExecuteReader()` calls `ExecuteReader(CommandBehavior.Default)` and there is not overload with a `string` parameter.

Either way, we can could end up creating two spans by instrumenting two of these methods in the call chain, which is why I added some code to skip creating a span if the active span is already a SQL Server span.

Reply:
It just occurred to me that we can instrument `ExecuteReader()` and `ExecuteReader(CommandBehavior)` and keep the de-duplication for cases when the former calls the latter. Should I change that?

Reply:
Yes, I think that is the best approach here 👍